tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -98,6 +135,57 @@ def __init__(self, logdir):
         """
         self._logdir = logdir
         self._reader = None
+        self._reader_lock = threading.Lock()
+        self._createOrReloadReader()
+
+    def _createOrReloadReader(self):
```

Comment:
It'd be helpful to break this method down since it's doing several different things and that makes it hard to reliably follow the details of the threading-related parts.  In particular it doesn't seem like it's necessary to obtain the reader lock if the reader already exists, because then you can just directly call `wake()/set()` to trigger an update.

FWIW, I'm also not entirely sure why it makes sense to trigger this logic from `Runs()` as well as from `__init__()`.  I guess because there's a chance the reader was not created due to a transient issue (the "no DebugEvent file set is found in the logdir") so we want to retry periodically?  But since this logic tightly couples both the reader creation and the immediate-update-triggering logic, it's not clear to me if `Runs()` also needs to trigger an immediate update of the reader (since the logic in `Runs()` doesn't actually use the reader directly, and it would be an asynchronous update anyway), or if that just sort of happens as an unnecessary side effect.

More comments about the expected lifecycle would help here, as would breaking this into more granular pieces that are explicitly called elsewhere in the class when needed.  For example:

```
def __init__(self):
    ...
    self._reader = None
    self._reader_lock = threading.Lock()
    self._monitors = []
    self._reload_needed_event = None
    self._tryCreateReader()
    self._tryReloadReader()

def _tryCreateReader(self):
    # Attempt to create the reader, using double-checked locking.
    if not self._reader:
        with self._reader_lock:
            if not self._reader:
                self._reader = self._createReader()
    # Reader creation may fail due to transient errors; abort and try later.
    if not self._reader:
        return
    self._monitors = self._createMonitors(self._reader)
    self._reload_needed_event = run_repeatedly_in_background(
        self._reader.update, DEFAULT_RELOAD_INTERVAL_SEC)

def _tryReloadReader(self)
    if self._reader:
        self._reload_needed_event.set()

def _createReader(self):
    ...

def _createMonitors(self, reader):
    ...
```

Reply:
Done breaking out these new methods:
- `_tryCreateReader()`
- `_reloadReader()`

I didn't break out additional new methods. The `self._monitor` is closedly related to the `self._reader`. So it's clearer to create the two in the same method, which is not overly long right now.

Re. the question about why calling `_tryCreateReader()` in the constructor and in `Runs()` I added comments in both places to clarify that. Briefly, the method is called in the constructor so the backend can start loading data as soon as the plugin backend is created, without waiting for the first request to come in from the FE. It's called in `Runs()` to cover the case in which the tfdbg2 data in the logdir is created after the tensorboard backend starts. I also added a few other comments to clarify the lifecycle of the reader.